lib/Frontend/Frontend.cpp

```diff
@@ -450,6 +450,16 @@ bool CompilerInstance::setUpModuleLoaders() {
     return true;
   }
 
+  // If implicit modules are disabled, we need to install an explicit module
+  // loader.
+  if (Invocation.getFrontendOptions().DisableImplicitModules) {
+    auto ESML = ExplicitSwiftModuleLoader::create(
+        *Context,
+        getDependencyTracker(), MLM,
+        Invocation.getSearchPathOptions().ExplicitSwiftModules,
+        IgnoreSourceInfoFile);
+    Context->addModuleLoader(std::move(ESML));
+  }
   if (MLM != ModuleLoadingMode::OnlySerialized) {
```

Comment:
When `ExplicitSwiftModuleLoader` is added, it seems like we should not add `ModuleInterfaceLoader` or `SerializedModuleLoader` at all. Then we don't need checks for `DisableImplicitModules` anywhere.